ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Compactor.java

```diff
@@ -28,6 +28,11 @@
 public interface Compactor {
 
   String FINAL_LOCATION = "hive.compactor.input.dir";
+  /*
+    Marker config key to indicate that the current Driver instance is used for executing compaction queries only.
+    It is required to exclude compaction related queries from all Ranger policies that would otherwise apply.
+   */
+  String COMPACTION_SESSION = "query_based_compactor_session";
```

Comment:
please follow property naming conventions like:
```
hive.compactor.query.based.session
```

Reply:
could the user use this directly on a query? e.g. doing this from beeline:
```
set query_based_compactor_session=true;
```
to see data unmasked even without permission to see? :) need to double-check

Reply:
hm, this was accidentally resolved I think, can you please address my second question?

Reply:
Yes, this can be done. I was thinking of it also, for safety I will replace this config value with a flag in `SessionState`.